optaplanner-docs/src/main/asciidoc/ConstraintStreams/ConstraintStreams-chapter.adoc

```diff
@@ -57,3 +57,86 @@ and supports <<explainingTheScore,justifications>>:
                         Talk::getDurationInMinutes);
     }
 ----
+
+[[constraintStreamsGroupingAndCollectors]]
+== Grouping and Collectors
+
+In the constraint streams framework, grouping is a mechanism to group items in a stream according to user-provider
```

Comment:
"In the constraint streams framework" -> it's a framework now? :)
I'd remove that context, as it's inside the ConstraintStreams session.

"grouping is a mechanism to group items in a stream according to ..."
Docs style suggestions: keep it short and use active voice, so instead:
"grouping collects items in a stream according to ..."
https://developers.google.com/style/voice